pipeline-model-api/src/main/java/org/jenkinsci/plugins/pipeline/modeldefinition/ast/ModelASTStage.java

```diff
@@ -112,7 +112,7 @@ public String toGroovy() {
                     result.append(',');
                 }
                 result.append('\n');
-                result.append(branch.getName()).append(": {\n").append(branch.toGroovy()).append("\n}");
+                result.append('"' + branch.getName() + '"').append(": {\n").append(branch.toGroovy()).append("\n}");
```

Comment:
🐛  defensively, I would also here (and any other places we're quoting things) replace quotes. e.g. `.replaceAll("\"", "\\\"")`; probably there's a utility function for this somewhere.